seurat/CellClassificationFXs.R

```diff
@@ -0,0 +1,605 @@
+SaveDimRedux_SERIII <- function(seuratObj, reductions=c("pca", "tsne", "umap"),
+                                save.path=NA, maxPCAcomps=10, nameID=""){
+
+  if(!is.na(save.path)){
+    require(data.table)
+
+    tempDT <- data.table(cbind(1:ncol(serObj), colnames(serObj)))
+    rownames(tempDT) <- colnames(serObj)
+    colnames(tempDT) <- c("nID", "cID")
+
+    tempDT <- merge(tempDT, )
+
+    if("tsne" %in% reductions) {
+      if(is.null(serObj@reductions$tsne)) print("tsne slot NULL") else {
+
+      }
+      tsneDT <- data.table(serObj@reductions$tsne@cell.embeddings)
+      tsneDT$cID <- rownames(serObj@reductions$tsne@cell.embeddings)
+      tempDT <- merge(tempDT, tsneDT, by="cID")
+    }
+    if("pca" %in% reductions) {
+      if(is.null(serObj@reductions$pca)) print("pca slot NULL") else {
+        pcaDT <- data.table(serObj@reductions$pca@cell.embeddings[,1:maxPCAcomps])
+        pcaDT$cID <- rownames(serObj@reductions$pca@cell.embeddings)
+        tempDT <- merge(tempDT, pcaDT, by="cID")
+      }
+
+    }
+    if("umap" %in% reductions) {
+      if(is.null(serObj@reductions$umap)) print("umap slot NULL") else {
+        umapDT <- data.table(serObj@reductions$umap@cell.embeddings)
+        umapDT$cID <- rownames(serObj@reductions$umap@cell.embeddings)
+        tempDT <- merge(tempDT, umapDT, by="cID")
+      }
+
+    }
+
+    if(!is.na(save.path)){
+
+      save.path <- paste(gsub(".csv", "", save.path), nameID, "_DimReduxComps.csv", sep="")
+      write.csv(tempDT, file = save.path, row.names=TRUE, col.names = T)
+
+    }
+
+
+  } else {
```

Comment:
i'm commenting out of linear order, but same as the ones below.  might as well put the sanity checking above and just return if it fails.